src/equations/ideal_glm_mhd_multiion.jl

```diff
@@ -0,0 +1,256 @@
+# This file includes functions that are common for all AbstractIdealGlmMhdMultiIonEquations
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+have_nonconservative_terms(::AbstractIdealGlmMhdMultiIonEquations) = True()
+
+function varnames(::typeof(cons2cons), equations::AbstractIdealGlmMhdMultiIonEquations)
+    cons = ("B1", "B2", "B3")
+    for i in eachcomponent(equations)
+        cons = (cons...,
+                tuple("rho_" * string(i), "rho_v1_" * string(i), "rho_v2_" * string(i),
+                      "rho_v3_" * string(i), "rho_e_" * string(i))...)
+    end
+    cons = (cons..., "psi")
+
+    return cons
+end
+
+function varnames(::typeof(cons2prim), equations::AbstractIdealGlmMhdMultiIonEquations)
+    prim = ("B1", "B2", "B3")
+    for i in eachcomponent(equations)
+        prim = (prim...,
+                tuple("rho_" * string(i), "v1_" * string(i), "v2_" * string(i),
+                      "v3_" * string(i), "p_" * string(i))...)
+    end
+    prim = (prim..., "psi")
+
+    return prim
+end
+
+function default_analysis_integrals(::AbstractIdealGlmMhdMultiIonEquations)
+    (entropy_timederivative, Val(:l2_divb), Val(:linf_divb))
+end
+
+"""
+    source_terms_lorentz(u, x, t, equations::AbstractIdealGlmMhdMultiIonEquations)
+
+Source terms due to the Lorentz' force for plasmas with more than one ion species. These source 
+terms are a fundamental, inseparable part of the multi-ion GLM-MHD equations, and vanish for 
+a single-species plasma. In particular, they have to be used for every
+simulation of [`IdealGlmMhdMultiIonEquations2D`](@ref).
+"""
+function source_terms_lorentz(u, x, t, equations::AbstractIdealGlmMhdMultiIonEquations)
+    @unpack charge_to_mass = equations
+    B1, B2, B3 = magnetic_field(u, equations)
+    v1_plus, v2_plus, v3_plus, vk1_plus, vk2_plus, vk3_plus = charge_averaged_velocities(u,
+                                                                                         equations)
+
+    s = zero(MVector{nvariables(equations), eltype(u)})
+
+    for k in eachcomponent(equations)
+        rho, rho_v1, rho_v2, rho_v3, rho_e = get_component(k, u, equations)
+        v1 = rho_v1 / rho
+        v2 = rho_v2 / rho
+        v3 = rho_v3 / rho
```

Comment:
`rho_inv` ?

Reply:
Thanks. I replaced all occurrences in ee33ce176a756cfc4571009b134dfeb841c21fde.